backend/src/app.ts

```diff
@@ -319,7 +319,48 @@ app.post('/api/add-like', authenticate, likeHandler(false));
 
 app.post('/api/remove-like', authenticate, likeHandler(true));
 
-app.put('/api/update-review-status/:reviewDocId/:newStatus', async (req, res) => {
+// Endpoint to update a review by its document ID
+app.put('/api/edit-review/:reviewDocId', async (req, res) => {
+  try {
+    const { reviewDocId } = req.params; // Extract the review document ID from the request parameters
+    const review = req.body as Review; // Get the review data from the request body and cast it to the "Review" type
+
+    // Check if the required fields (overallRating and reviewText) are missing or invalid
+    if (review.overallRating === 0 || review.reviewText === '') {
+      res.status(401).send('Error: missing fields');
+    }
+
+    // Update the review document in the database with the provided data
+    await reviewCollection
+      .doc(reviewDocId)
+      .update({ ...review, date: new Date(review.date), status: 'PENDING' });
```

Comment:
Are you leaving the old date or is this updating it with a new date/do you know which one it should be?